### PR TITLE
MULE-18753: Add live Sample Data fetching capabilities

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/MuleSampleDataService.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  private MuleContext muleContext;
+  private StreamingManager streamingManager;
+  private ConnectionManager connectionManager;
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Message getSampleData(Location location) throws SampleDataException {
+    boolean isConnection = isConnection(location);
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/MuleSampleDataService.java:65]: Should we take this into account? only operation and sources locations are valid here.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/SampleDataProviderFactory.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  <T, A> SampleDataProvider<T, A> createSampleDataProvider() throws SampleDataException {
+    Class<? extends SampleDataProvider<T, A>> providerClass = factoryModelProperty.getSampleDataProviderClass();
+
+    try {
+      SampleDataProvider<T, A> resolver = instantiateClass(providerClass);
+      initialiseIfNeeded(resolver, true, muleContext);
+
+      injectValueProviderFields(resolver);
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/SampleDataProviderFactory.java:76]: injectSampleDataProviderFields?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/SampleDataProviderMediator.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import org.mule.sdk.api.data.sample.SampleDataProvider;
+import org.mule.sdk.api.runtime.operation.Result;
+
+import java.util.function.Supplier;
+
+/**
+ * Coordinates all the moving parts necessary to provision and execute a {@link SampleDataProvider},
+ * handling possible errors and transforming the output into a {@link Message}
+ *
+ * @since 4.4.0SampleDataDeclarationEnricher
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/SampleDataProviderMediator.java:47]: delete SampleDataDeclarationEnricher

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/data/sample/SampleDataTestUtils.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * Utilities for testing {@link SampleDataProvider} functionalities
+ *
+ * @since 4.4.0
+ */
+public final class SampleDataTestUtils {
+
+  private SampleDataTestUtils() {}
+
+  /**
+   * Creaes a matcher that tests that a {@link SampleDataException} has the given {@code failureCode}
```
Comment (ndinu) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/data/sample/SampleDataTestUtils.java:26]: Creates

**Review by ndinu — CHANGES_REQUESTED** [review]

**Comment by marianogonzalez** [comment]
--validate

**Review by ndinu — APPROVED** [review]

**Pull request merged**